### ISSUE TITLE
Throttling API calls due to errors with the 10 previous attempts. I'll retry again in 5 minutes.

### ISSUE THREAD

**Issue by jcleek** [hjdhjd/homebridge-unifi-protect#821]
**Describe The Problem:**
[31/10/2022, 12:57:44] [UniFi Protect] 10.0.2.2: Throttling API calls due to errors with the 10 previous attempts. I'll retry again in 5 minutes.
[31/10/2022, 13:02:44] [UniFi Protect] 10.0.2.2: Resuming connectivity to the UniFi Protect API after throttling for 5 minutes.
[31/10/2022, 13:03:15] [UniFi Protect] 10.0.2.2: Throttling API calls due to errors with the 10 previous attempts. I'll retry again in 5 minutes.

I have created new admin users, both local and Unifi accounts. I have uninstalled and reinstalled the plugin. I have re-written the config multiple times. I have restarted both the UDM-Pro and UNVR. All OS updates have been applied on my Ubiquiti equipment as well as the Raspberry Pi running Homebridge. I have removed non-Homebridge or Protect messages from logs presented.

**To Reproduce:**
Start Homebridge with the Unifi Protect plugin installed

**Logs:**
^[[39m 10.0.2.2: Throttling API calls due to errors with the 10 previous attempts. I'll retry again in 5 mi$
^[[37m[31/10/2022, 04:14:09] ^[[39m^[[36m[UniFi Protect]^[[39m 10.0.2.2: Resuming connectivity to the UniFi$
^[[37m[31/10/2022, 04:14:40] ^[[39m^[[36m[UniFi Protect]^[[39m 10.0.2.2: Throttling API calls due to errors$
^[[37m[31/10/2022, 04:19:40] ^[[39m^[[36m[UniFi Protect]^[[39m 10.0.2.2: Resuming connectivity to the UniFi$
^[[37m[31/10/2022, 04:20:11] ^[[39m^[[36m[UniFi Protect]^[[39m 10.0.2.2: Throttling API calls due to errors$
^[[37m[31/10/2022, 04:25:11] ^[[39m^[[36m[UniFi Protect]^[[39m 10.0.2.2: Resuming connectivity to the UniFi$
^[[37m[31/10/2022, 04:25:41] ^[[39m^[[36m[UniFi Protect]^[[39m 10.0.2.2: Throttling API calls due to errors$
^[[37m[31/10/2022, 04:30:41] ^[[39m^[[36m[UniFi Protect]^[[39m 10.0.2.2: Resuming connectivity to the UniFi$

```
Show the Homebridge logs here.
[31/10/2022, 12:57:06] [HB Supervisor] Starting Homebridge with extra flags: -I
[31/10/2022, 12:57:06] [HB Supervisor] Started Homebridge v1.5.1 with PID: 4402
[31/10/2022, 12:57:07] Loaded config.json with 6 accessories and 7 platforms.
[31/10/2022, 12:57:08] Loaded 44 cached accessories from cachedAccessories.
[31/10/2022, 12:57:08] ---
[31/10/2022, 12:57:10] Loaded plugin: homebridge-config-ui-x@4.50.1
[31/10/2022, 12:57:10] Registering platform 'homebridge-config-ui-x.config'
[31/10/2022, 12:57:12] ---
[31/10/2022, 12:57:13] Loaded plugin: homebridge-unifi-protect@5.5.4
[31/10/2022, 12:57:13] Registering platform 'homebridge-unifi-protect.UniFi Protect'
[31/10/2022, 12:57:13] ---
[31/10/2022, 12:57:13] Loading 7 platforms...
[31/10/2022, 12:57:13] [UniFi Protect] Initializing UniFi Protect platform...
[31/10/2022, 12:57:13] Loading 6 accessories...
Setup Payload:
X-HM://0024E8RLF8WB3
Enter this code with your HomeKit app on your iOS device to pair with Homebridge:
                       
    ┌────────────┐     
    │ 000-00-000 │     
    └────────────┘     
                       
[31/10/2022, 12:57:13] Homebridge v1.5.1 (HAP v0.10.4) (Homebridge) is running on port 51683.
[31/10/2022, 12:57:44] [UniFi Protect] 10.0.2.2: Throttling API calls due to errors with the 10 previous attempts. I'll retry again in 5 minutes.
[31/10/2022, 13:02:44] [UniFi Protect] 10.0.2.2: Resuming connectivity to the UniFi Protect API after throttling for 5 minutes.
[31/10/2022, 13:03:15] [UniFi Protect] 10.0.2.2: Throttling API calls due to errors with the 10 previous attempts. I'll retry again in 5 minutes.

**Homebridge Configuration:**
{
    "bridge": {
        "name": "Homebridge",
        "username": "00:00:00:00:00:00",
        "port": 12345,
        "pin": "000-00-000"
    },
    ],
    "platforms": [
        {
            "name": "Config",
            "port": 8581,
            "auth": "none",
            "theme": "dark-mode",
            "tempUnits": "f",
            "lang": "auto",
            "platform": "config"
        },
        {
            "controllers": [
                {
                    "address": "10.0.2.2",
                    "password": "xxxxxxxxx",
                    "username": "Homekit"
                }
            ],
            "platform": "UniFi Protect"
        },
    ],
    "disabledPlugins": []
}

**Screenshots:**
Same as errors posted above and offer no additional value

**Environment:**

* **Homebridge Version:** 1.5.1
* **Node Version:** 16.15.0
* **Homebridge-UniFi-Protect Plugin Version:** 5.5.4
* **UniFi Protect Version:** 2.2.6
* **UniFi Protect Controller Type:** UNVR
* **UniFi Protect Controller Firmware Version:** 2.5.11
* **Apple Device and iOS / macOS / iPadOS / tvOS Version:** Not relevant
* **Operating System and OS Version:** Raspbian GNU/Linux Buster (10)

**Comment by jcleek** [comment]
I tried the plugin called "Homebridge Unifi Protect Platform" (https://www.npmjs.com/package/homebridge-unifi-protect-platform) just to see if I got different results. I was not able to get it to work but using the same local account I have set up for this plugin I was able to successfully pull a list of my cameras. The plugin itself would not register so I couldn't see anything but I know the account works, and that other plugins can get onto the current version of Protect on the UNVR.

**Comment by hjdhjd** [comment]
The issue you've got looks like it's related to not having a local user properly configured, I suspect. Please post screenshots showing the local user and the roles you've assigned. Thanks.

**Comment by jcleek** [comment]
![image](https://user-images.githubusercontent.com/16121417/199805829-31f94938-9e0d-4c58-916c-41643be658ea.png)

**Comment by jcleek** [comment]
They are an administrator of the NVR.

**Comment by jcleek** [comment]
Also the plugin was working, until it stopped with no changes made. In trying to get it to work again I have created and deleted local and Ubiquiti accounts many times. Every time they have been created with the administrator role.

**Comment by hjdhjd** [comment]
Did you happen to upgrade your Node version perhaps?

I'd also try uninstalling and reinstalling the plugin, just to ensure there's nothing strange when it comes to dependencies as can happen occasionally. At a bit of a loss otherwise, I'm afraid.

**Comment by jcleek** [comment]
I wiped my Homebridge install and started fresh, using the Raspberry Pi imager to install. After the reboot I was prompted to update homebridge and the Homebridge UI plugin which I did. Once that was complete I installed your plugin and set it up. I get the same errors. I have a fresh install, with only your plugin added.

So I created a new ubiquiti account on my NVR, updated the config and restarted. Same errors.

To recap: new Homebridge install. Only your plug-in installed. Tried both local and ubiquiti accounts for my NVR. Both failed. This was an install that had been working for well over a year. There is nothing else I can try as I have replaced every possible piece on my end besides the NVR.

**Comment by jcleek** [comment]
![image](https://user-images.githubusercontent.com/16121417/200025057-193f345a-3a52-4c66-a0e9-724c8036db6b.png)

**Comment by jcleek** [comment]
![image](https://user-images.githubusercontent.com/16121417/200025219-c7d78852-9e07-4ed3-b0d8-0fca8ef52da2.png)

**Comment by jcleek** [comment]
![image](https://user-images.githubusercontent.com/16121417/200025290-5ce09f76-736c-42b9-9f3b-554b5def29d0.png)

**Comment by jcleek** [comment]
![image](https://user-images.githubusercontent.com/16121417/200025404-2206fe69-6612-4206-bca4-ddddebc5bbbd.png)

**Comment by jcleek** [comment]
I tried setting the account to administrator, user and member. None of them worked with either a ubiquiti or local account.

**Comment by jcleek** [comment]
Problem solved and completely plug-in unrelated.

**Comment by hjdhjd** [comment]
What was the issue, for future reference for others?

**Comment by jcleek** [comment]
I'm stupid. Nobody needs to know that.

I recently upgraded my switch to a Pro which supports 10Gb through the SFP. So I got SFPs and connected them in place of the ethernet connection. Unfortunately I did not realize the IP reservation would not follow the UNVR with the new connection and it pulled a different IP.

**Comment by hjdhjd** [comment]
Nah…just human. We tend to overlook the simple stuff sometimes…happens to us all. Plus, we have confirmation bias…so if you see an issue that kind of looks like what you think it might be…you start seeing it everywhere. Glad it worked out. Enjoy!
